tests/cross/src/test/scala/tests/pc/CompletionSnippetSuite.scala

```diff
@@ -439,4 +439,29 @@ class CompletionSnippetSuite extends BaseCompletionSuite {
     includeDetail = true
   )
 
+  // https://github.com/scalameta/metals/issues/4004
+  checkSnippet(
+    "issu-4004".tag(IgnoreScala2),
+    s"""|package a
+        |object Foo:
+        |  extension (s: String)
+        |    def barr = 0
+        |    def baxx() = 0
+        |    def bazz()() = 0
+        |  val bar = "abc".bar@@
+        |  val bax = "abc".bax@@
```

Comment:
I don't the tests will correctly work with multiple `@@` I think you need to create 3 separate tests

Reply:
Also, check snippet only returns the completion item. You would need to do checkEdit  to get the full output I think.

You can run the tests with `sbt` -> `++3.1.2` -> "cross/testOnly tests.pc.CompletionSnippetSuite"